### ISSUE TITLE
main.py

### ISSUE THREAD

**Issue by Hyunji-K1m** [Hyunji-K1m/Hyunji-K1m.github.io#2]
import pandas as pd
from sklearn.preprocessing import StandardScaler
import numpy as np
from copy import deepcopy
import torch
from torch.utils.data import Dataset, DataLoader
import sys
import os

sys.path.append(os.path.abspath('/Users/kimhyunji/Desktop/Dissertation/code/ready'))

import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import torch
import torch.optim as optim
from torch.utils.data import DataLoader
from LSTNet import LSTNet, TimeSeriesDataset, smape, mase

# Load the dataset
df = pd.read_csv('/Users/kimhyunji/Desktop/Dissertation/code/data/electricity consumption_daily.csv', index_col=0)
df = df.reset_index()

pivot_df = df.pivot(index='start_date', columns='series_name', values='daily_series_value')
data = pivot_df.values

# 데이터 스케일링
scaler = StandardScaler()
data = scaler.fit_transform(data)

train_data, test_data = train_test_split(data, train_size=0.8, shuffle=True, random_state=42)

# 파라미터 설정
input_size = train_data.shape[1]
hidden_size = 64
output_size = train_data.shape[1]
seq_len = 10
batch_size = 32
num_epochs = 10
learning_rate = 0.001

# 데이터 로더 설정
train_dataset = TimeSeriesDataset(train_data, seq_len)
test_dataset = TimeSeriesDataset(test_data, seq_len)
train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
test_loader = DataLoader(test_dataset, batch_size=batch_size, shuffle=False)

# 모델, 손실 함수, 옵티마이저 설정
model = LSTNet(input_size, hidden_size, output_size)
criterion = torch.nn.MSELoss()
optimizer = optim.Adam(model.parameters(), lr=learning_rate)

for epoch in range(num_epochs):
    model.train()
    for inputs, targets in train_loader:
        inputs, targets = torch.tensor(inputs, dtype=torch.float32), torch.tensor(targets, dtype=torch.float32)
        optimizer.zero_grad()
        outputs = model(inputs)
        loss = criterion(outputs, targets)
        loss.backward()
        optimizer.step()
    print(f'Epoch {epoch+1}/{num_epochs}, Loss: {loss.item()}')
